packages/app/src/systems/Sidebar/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+import { cssObj } from "@fuel-ui/css";
+import type { Icons } from "@fuel-ui/react";
+import { Box, Flex, Icon, Menu as RootMenu } from "@fuel-ui/react";
+import { AnimatePresence, motion } from "framer-motion";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+export type MenuItemObj = {
+  key: string;
+  icon: Icons;
+  label: string;
+  path?: string;
+  ahref?: string;
+  submenu?: MenuItemObj[];
+};
+
+type MenuItemContentProps = {
+  item: MenuItemObj;
+  isOpened?: boolean;
+};
+
+function MenuItemContent({ item, isOpened }: MenuItemContentProps) {
+  const caret = isOpened ? Icon.is("CaretUp") : Icon.is("CaretDown");
+  const navigate = useNavigate();
+
+  function handleAction(key: string | number) {
+    const subItem = item.submenu?.find((i) => i.key === key);
+    if (subItem?.path) {
+      navigate(subItem.path);
+    }
+    if (subItem?.ahref) {
+      window.open(subItem.ahref);
+    }
+  }
+
+  return (
+    <Flex direction="column" css={styles.menuItemContent}>
+      <Flex gap="$3">
+        <Icon
+          icon={item.icon}
+          css={{ color: "$gray8" }}
+          className="main-icon"
+        />
+        <Box css={{ flex: 1 }}>{item.label}</Box>
+        {item.submenu && <Icon icon={caret} css={{ color: "$gray8" }} />}
+      </Flex>
+      <AnimatePresence>
+        {item.submenu && isOpened && (
+          <motion.div
+            initial={{ opacity: 0, y: -30 }}
+            animate={{ opacity: 1, y: 0 }}
+            exit={{ opacity: 0, y: -30 }}
+            transition={{ ease: "linear", duration: "0.1" }}
+          >
+            <RootMenu
+              css={styles.submenu}
+              onAction={handleAction}
+              aria-label="Submenu"
+            >
+              {item.submenu.map((subItem) => (
+                <RootMenu.Item key={subItem.key} textValue={subItem.label}>
+                  <Icon icon={subItem.icon} css={{ color: "$gray8" }} />
+                  {subItem.label}
+                </RootMenu.Item>
+              ))}
+            </RootMenu>
+          </motion.div>
+        )}
+      </AnimatePresence>
+    </Flex>
+  );
+}
+
+export type MenuProps = {
+  items: MenuItemObj[];
+};
+
+export function Menu({ items }: MenuProps) {
+  const [opened, setOpened] = useState<string | null>(null);
+  const navigate = useNavigate();
+
+  function handleAction(key: string | number) {
```

Comment:
we can remove the tslint comment in beggining of this file.

changing this function to:


```
function handleAction(key: string | number) {
    const item = items.find((item) => item.key === key);
    if (item?.submenu) {
      setOpened(opened !== item.key ? item.key : null);
    } else if (item?.path) {
      navigate(item.path);
    }
  }
```

Reply:
✅ @LuizAsFight